### PR TITLE
rustbuild: per target musl-root

### PR DESCRIPTION
**Pull request by japaric**
config.toml now accepts a target.$TARGET.musl-root key that lets you
override the "build" musl-root value, which is set via the --musl-root
flag or via the build.musl-root key.

With this change, it's now possible to compile std for several musl
targets at once. Here's are the sample commands to do such thing:

```
$ configure \
    --enable-rustbuild \
    --target=x86_64-unknown-linux-musl,arm-unknown-linux-musleabi \
    --musl-root=/musl/x86_64-unknown-linux-musl/

$ edit config.toml && tail config.toml
[target.arm-unknown-linux-musleabi]
musl-root = "/x-tools/arm-unknown-linux-musleabi/arm-unknown-linux-musleabi/sysroot/usr"

$ make
```

r? @alexcrichton 
With this we should be able to start producing releases of std for arm musl targets

### CODE REVIEW

File: src/bootstrap/sanity.rs [hunk 1]
```diff
@@ -111,8 +111,9 @@ pub fn check(build: &mut Build) {
 
         // Make sure musl-root is valid if specified
         if target.contains("musl") && !target.contains("mips") {
-            match build.config.musl_root {
-                Some(ref root) => {
+            match build.config.target_config[target].musl_root.as_ref()
+                .or(build.config.musl_root.as_ref()) {
```
Comment (alexcrichton) [inline src/bootstrap/sanity.rs]: Perhaps this logic with `.or` could be a helper method on `Build`?

**Comment by alexcrichton** [comment]
Nice! Could you also update `config.toml.example` with a comment about this option? You can also remove the top-level `musl-root` in the example file as it basically shouldn't be used any more.

**Comment by alexcrichton** [comment]
(remove it from the documentation, that is)

**Comment by japaric** [comment]
@alexcrichton addressed your comments

**Pull request merged**